chemprop/data/data.py

```diff
@@ -67,42 +69,55 @@ def __init__(self,
                  atom_features: np.ndarray = None,
                  atom_descriptors: np.ndarray = None,
                  bond_features: np.ndarray = None,
+                 bond_descriptors: np.ndarray = None,
+                 constraints: np.ndarray = None,
+                 raw_constraints: np.ndarray = None,
                  overwrite_default_atom_features: bool = False,
                  overwrite_default_bond_features: bool = False):
         """
         :param smiles: A list of the SMILES strings for the molecules.
         :param targets: A list of targets for the molecule (contains None for unknown target values).
+        :param atom_targets: A list of targets for the atomic properties.
+        :param bond_targets: A list of targets for the bond properties.
         :param row: The raw CSV row containing the information for this molecule.
         :param data_weight: Weighting of the datapoint for the loss function.
         :param gt_targets: Indicates whether the targets are an inequality regression target of the form ">x".
         :param lt_targets: Indicates whether the targets are an inequality regression target of the form "<x".
         :param features: A numpy array containing additional features (e.g., Morgan fingerprint).
         :param features_generator: A list of features generators to use.
         :param phase_features: A one-hot vector indicating the phase of the data, as used in spectra data.
-        :param atom_descriptors: A numpy array containing additional atom descriptors to featurize the molecule
-        :param bond_features: A numpy array containing additional bond features to featurize the molecule
-        :param overwrite_default_atom_features: Boolean to overwrite default atom features by atom_features
-        :param overwrite_default_bond_features: Boolean to overwrite default bond features by bond_features
+        :param atom_descriptors: A numpy array containing additional atom descriptors to featurize the molecule.
+        :param bond_descriptors: A numpy array containing additional bond descriptors to featurize the molecule.
+        :param constraints: A numpy array containing constraints applied to different atomic/bond properties.
+        :param raw_constraints: A numpy array containing constraints in raw data.
```

Comment:
`raw_constraints` is a numpy array that contains every constraint presented in row data, even if the corresponding columns are not used as targets. On the other hand, `constraints` contains only a subset of raw_constraints corresponding to the targets. The reason to preserve `raw_constraints` is because the raw constraints for each split can be saved when `--save_smiles_splits` flag is used.

Reply:
Thanks. For clarity, maybe consider revising the parameter description to:

`
:param raw_constraints: A numpy array containing all user-provided atom/bond-level constraints in input data.
`
`
:param constraints: A numpy array containing atom/bond-level constraints that are used in training. Param constraints is a subset of param raw_constraints.
`